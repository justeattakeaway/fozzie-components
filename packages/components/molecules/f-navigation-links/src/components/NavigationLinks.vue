<template>
    <ul
        v-if="links.length"
        :class="$style['c-navigationLinks']"
        data-test-id="navigationLinks">
        <li
            v-for="({ id, url, name, selected }, i) in links"
            :key="i"
            :class="$style['c-navigationLinks-item']">
            <span
                v-if="selected"
                :data-test-id="id"
                tabindex="0"
                :class="[$style['c-navigationLinks-link'], $style['c-navigationLinks-link--active']]">
                {{ name }}
            </span>
            <v-link
                v-else
                :data-test-id="id"
                :has-text-decoration="false"
                :href="url"
                :class="$style['c-navigationLinks-link']">
                {{ name }}
            </v-link>
        </li>
    </ul>
</template>

<script>
import VLink from '@justeat/f-link';
import '@justeat/f-link/dist/f-link.css';

export default {
    name: 'NavigationLinks',

    components: {
        VLink
    },

    props: {
        links: {
            type: Array,
            default: () => []
        }
    }
};
</script>

<style lang="scss" module>
.c-navigationLinks {
    list-style-type: none;
    margin: 0;
    padding: 0;
}

.c-navigationLinks-item {
    display: inline-block;
    width: 100%;
}

.c-navigationLinks-link {
    display: block;
    padding: spacing() 0 spacing() spacing(x2);
    border-left: 2px solid $color-grey-30;
    color: $color-grey;

    &:focus, &:hover {
        border-left: 2px solid $color-orange;
        text-decoration: none;
    }
}

.c-navigationLinks-link--active {
    border-left: 2px solid $color-orange;
    font-weight: $font-weight-bold;
}

</style>
