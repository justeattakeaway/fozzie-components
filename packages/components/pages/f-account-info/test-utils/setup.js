import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
