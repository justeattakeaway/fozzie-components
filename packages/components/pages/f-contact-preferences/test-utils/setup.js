export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
