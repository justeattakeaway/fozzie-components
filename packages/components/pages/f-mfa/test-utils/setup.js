export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
