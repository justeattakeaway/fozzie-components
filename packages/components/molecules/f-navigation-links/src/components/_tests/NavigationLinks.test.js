import { shallowMount, RouterLinkStub } from '@vue/test-utils';

import NavigationLinks from '../NavigationLinks.vue';

const propsData = {
    links: [
        {
            id: 'link1',
            to: '/account/info',
            name: 'Your account'
        },
        {
            id: 'link2',
            to: '/order-history',
            name: 'Your orders'
        },
        {
            id: 'link3',
            to: '/account/paymentoptions',
            name: 'Your saved cards'

        },
        {
            id: 'link4',
            to: '/account/addressbook',
            name: 'Your address book'
        },
        {
            id: 'link5',
            to: '/account/credit',
            name: 'Redeem a voucher'

        },
        {
            id: 'link6',
            to: '/giftcards/redeem',
            name: 'Redeem a gift card'
        },
        {
            id: 'link7',
            to: '/account/preferences',
            name: 'Contact preferences'
        },
        {
            id: 'link8',
            href: '/anchor-link',
            name: 'Anchor link'
        }
    ]
};
const componentTag = "[data-test-id='navigation-links-component']";

describe('NavigationLinks', () => {
    it('should be defined', () => {
        // Arrange & Act
        const wrapper = shallowMount(NavigationLinks, {
            propsData,
            stubs: {
                RouterLink: RouterLinkStub
            }
        });

        // Assert
        expect(wrapper.exists()).toBe(true);
    });

    it('should not fail if no `links` prop supplied', () => {
        // Arrange & Act
        const wrapper = shallowMount(NavigationLinks, {
            stubs: {
                RouterLink: RouterLinkStub
            }
        });

        // Assert
        expect(wrapper.html).toMatchSnapshot();
    });

    it('should display the list elements correctly for each `link` supplied', () => {
        // Arrange
        const wrapper = shallowMount(NavigationLinks, {
            propsData,
            stubs: {
                RouterLink: RouterLinkStub
            }
        });

        // Act
        const links = wrapper.find(componentTag);

        // Assert
        expect(links).toMatchSnapshot();
    });

    it('should pass to attribute to `<v-link>` when present', () => {
        // Arrange
        const expected = '/account/info';
        const wrapper = shallowMount(NavigationLinks, {
            propsData: {
                links: [
                    {
                        id: 'link1',
                        to: expected,
                        name: 'Your account'
                    }
                ]
            },
            stubs: {
                RouterLink: RouterLinkStub
            }
        });

        // Act
        const link = wrapper.find('[data-test-id="link1"]');

        // Assert
        expect(link.attributes('to')).toEqual(expected);
    });

    it('should pass href attribute to `<v-link>` when present', () => {
        // Arrange
        const expected = '/account/info';
        const wrapper = shallowMount(NavigationLinks, {
            propsData: {
                links: [
                    {
                        id: 'link1',
                        href: expected,
                        name: 'Your account'
                    }
                ]
            },
            stubs: {
                RouterLink: RouterLinkStub
            }
        });

        // Act
        const link = wrapper.find('[data-test-id="link1"]');

        // Assert
        expect(link.attributes('href')).toEqual(expected);
    });

    it('should pass href attribute to `<v-link>` when both `to` and `href` present in links data', () => {
        // Arrange
        const expected = '/account/info';
        const wrapper = shallowMount(NavigationLinks, {
            propsData: {
                links: [
                    {
                        id: 'link1',
                        href: expected,
                        to: '/not-used',
                        name: 'Your account'
                    }
                ]
            },
            stubs: {
                RouterLink: RouterLinkStub
            }
        });

        // Act
        const link = wrapper.find('[data-test-id="link1"]');

        // Assert
        expect(link.attributes('href')).toEqual(expected);
        expect(link.attributes('to')).toBeUndefined();
    });

    it('should skip link item if both `to` and `href` missing in props.links object', () => {
        // Arrange
        const wrapper = shallowMount(NavigationLinks, {
            propsData: {
                links: [
                    {
                        id: 'link1',
                        name: 'Your account'
                    },
                    {
                        id: 'link2',
                        to: '/account/info',
                        name: 'Your account'
                    }
                ]
            },
            stubs: {
                RouterLink: RouterLinkStub
            }
        });

        // Act
        const links = wrapper.find(componentTag);
        const link = wrapper.find('[data-test-id="link1"]');

        // Assert
        expect(links.findAll('li').length).toEqual(1);
        expect(link.exists()).toBeFalsy();
    });

    it('should contain `aria-label` with content `navigation`', () => {
        // Arrange
        const wrapper = shallowMount(NavigationLinks);

        // Act
        const links = wrapper.find(componentTag);

        // Assert
        expect(links.attributes('aria-label')).toBe('navigation');
    });

    describe('computed ::', () => {
        describe('filterLinks ::', () => {
            it('should filter props.links if link does not have `to` or `href` property', () => {
                // Arrange
                const wrapper = shallowMount(NavigationLinks, {
                    propsData: {
                        links: [
                            {
                                id: 'link1',
                                name: 'Link 1'
                            },
                            {
                                id: 'link2',
                                to: '/account/info',
                                name: 'Your account'
                            },
                            {
                                id: 'link3',
                                name: 'Link 3',
                                href: '/link-3'
                            }
                        ]
                    },
                    stubs: {
                        RouterLink: RouterLinkStub
                    }
                });

                // Act
                const result = wrapper.vm.filterLinks;

                // Assert
                expect(result.length).toEqual(2);
                expect(result.find(x => x.id === 'link1')).toBeUndefined();
            });
        });
    });
});
