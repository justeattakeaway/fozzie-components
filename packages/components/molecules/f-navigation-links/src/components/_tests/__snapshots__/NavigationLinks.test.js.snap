// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLinks should display the list elements correctly for each \`link\` supplied 1`] = `
<nav aria-label="navigation" data-test-id="navigation-links-component">
  <ul>
    <li>
      <v-link-stub data-test-id="link1" link-class="c-navigationLinks-link" to="/account/info">
        Your account
      </v-link-stub>
    </li>
    <li>
      <v-link-stub data-test-id="link2" link-class="c-navigationLinks-link" to="/order-history">
        Your orders
      </v-link-stub>
    </li>
    <li>
      <v-link-stub data-test-id="link3" link-class="c-navigationLinks-link" to="/account/paymentoptions">
        Your saved cards
      </v-link-stub>
    </li>
    <li>
      <v-link-stub data-test-id="link4" link-class="c-navigationLinks-link" to="/account/addressbook">
        Your address book
      </v-link-stub>
    </li>
    <li>
      <v-link-stub data-test-id="link5" link-class="c-navigationLinks-link" to="/account/credit">
        Redeem a voucher
      </v-link-stub>
    </li>
    <li>
      <v-link-stub data-test-id="link6" link-class="c-navigationLinks-link" to="/giftcards/redeem">
        Redeem a gift card
      </v-link-stub>
    </li>
    <li>
      <v-link-stub data-test-id="link7" link-class="c-navigationLinks-link" to="/account/preferences">
        Contact preferences
      </v-link-stub>
    </li>
    <li>
      <v-link-stub data-test-id="link8" link-class="c-navigationLinks-link" href="/anchor-link">
        Anchor link
      </v-link-stub>
    </li>
  </ul>
</nav>
`;

exports[`NavigationLinks should not fail if no \`links\` prop supplied 1`] = `[Function]`;
